src/main/java/minegame159/meteorclient/systems/modules/combat/BowAimbot.java

```diff
@@ -0,0 +1,181 @@
+import net.minecraft.entity.EntityType;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.passive.AnimalEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.BowItem;
+import net.minecraft.item.Items;
+import net.minecraft.util.math.Vec3d;
+
+public class BowAimbot extends Module {
+    private final SettingGroup sgGeneral = settings.getDefaultGroup();
+
+    private final Setting<Double> range = sgGeneral.add(new DoubleSetting.Builder()
+            .name("range")
+            .description("The maximum range the entity can be to aim at it.")
+            .defaultValue(20)
+            .min(0)
+            .max(100)
+            .sliderMax(100)
+            .build()
+    );
+
+    private final Setting<Object2BooleanMap<EntityType<?>>> entities = sgGeneral.add(new EntityTypeListSetting.Builder()
+            .name("entities")
+            .description("Entities to attack.")
+            .defaultValue(new Object2BooleanOpenHashMap<>(0))
+            .onlyAttackable()
+            .build()
+    );
+
+    private final Setting<SortPriority> priority = sgGeneral.add(new EnumSetting.Builder<SortPriority>()
+            .name("priority")
+            .description("What type of entities to target.")
+            .defaultValue(SortPriority.LowestHealth)
+            .build()
+    );
+
+    private final Setting<Boolean> friends = sgGeneral.add(new BoolSetting.Builder()
+            .name("friends")
+            .description("Whether or not to attack friends. Useful if you select players selected.")
+            .defaultValue(false)
+            .build()
+    );
+
+    private final Setting<Boolean> babies = sgGeneral.add(new BoolSetting.Builder()
+            .name("babies")
+            .description("Whether or not to attack baby variants of the entity.")
+            .defaultValue(true)
+            .build()
+    );
+
+    private final Setting<Boolean> nametagged = sgGeneral.add(new BoolSetting.Builder()
+            .name("nametagged")
+            .description("Whether or not to attack mobs with a name tag.")
+            .defaultValue(false)
+            .build()
+    );
+
+    private final Setting<Boolean> pauseOnCombat = sgGeneral.add(new BoolSetting.Builder()
+            .name("pause-on-combat")
+            .description("Freezes Baritone temporarily until you released the bow.")
+            .defaultValue(false)
+            .build()
+    );
+
+    private boolean wasPathing;
+    private Entity target;
+
+    public BowAimbot() {
+        super(Categories.Combat, "Bow Aimbot", "Automatically aims your bow for you.");
+    }
+
+    @Override
+    public void onDeactivate() {
+        target = null;
+        wasPathing = false;
+    }
+
+    @EventHandler
+    private void onTick(TickEvent.Pre event) {
+        if (mc.player.isDead() || !mc.player.isAlive() || !itemInHand()) return;
+
+        // Target
+        if (InvUtils.findItemWithCount(Items.ARROW).slot == -1) return;
```

Comment:
I think you should be using `InvUtils.findItemInHotbar` for this check since `findItemWithCount` searches your whole inv iirc.

Reply:
Wouldn't that mean you have to have the arrows be in your hotbar for it to work?
Imo it would be annoying to have to place the arrows into the hotbar everytime.

Also, I was following the BowSpam module for that.
```
    private void onTick(TickEvent.Post event) {
        if (InvUtils.findItemWithCount(Items.ARROW).slot == -1) return;
        if (!onlyWhenHoldingRightClick.get() || mc.options.keyUse.isPressed()) {
```

Reply:
i shouldn't comment just after i wake up, sorry